lib/v2/facebook/page.js

```diff
@@ -0,0 +1,259 @@
+    }
+
+    return {
+        url,
+        title,
+        content,
+        images,
+    };
+};
+
+module.exports = async (ctx) => {
+    const { id } = ctx.params;
+    const pageId = encodeURIComponent(id);
+
+    // try fetch from mobile site
+    const html = await fetchPageHtml('m', `/${pageId}`, pageId, ctx.cache);
+    const $ = cheerio.load(html);
+
+    const $mobileHeader = $('#msite-pages-header-contents');
+
+    if ($mobileHeader.length > 0) {
+        const title = $mobileHeader.find('div[data-nt="FB:TEXT4"]').eq(0).text();
+        const description = $('#pages_msite_body_contents div[data-nt="FB:EXPANDABLE_TEXT"]').text();
+        const htmlPosts = await fetchPageHtml('m', `/${pageId}/posts`, `${pageId}_posts`, ctx.cache);
+        const $posts = cheerio.load(htmlPosts);
+
+        const itemLinks = $posts('div.story_body_container > div[data-ft=\'{"tn":"*s"}\'] > a:last-child')
+            .toArray()
+            .map((a) => $(a).attr('href'));
+
+        const items = await Promise.all(
+            itemLinks.map(async (itemLink) => {
+                if (new RegExp(`^/.+/photos/`).test(itemLink)) {
+                    const data = await parsePhotoPage(itemLink, ctx.cache);
+                    return {
+                        title: data.title,
+                        link: data.url,
+                        description: `<img src="${data.image}"><br>${data.content}`,
+                    };
+                }
+                if (new RegExp(`^/story.php`).test(itemLink)) {
+                    const data = await parseStoryPage(itemLink, ctx.cache);
+                    const isEmptyImageList = data.images.length === 0;
+
+                    let desc = data.content;
+                    if (!isEmptyImageList) {
+                        desc += '<br>' + data.images.map((image) => `<img src="${image.image}"><br>${image.content}`).join('<br>');
+                    }
+
+                    return {
+                        title: data.title,
+                        link: data.url,
+                        description: desc,
+                    };
+                }
+            })
+        );
+
+        ctx.state.data = {
+            title,
+            link: `https://www.facebook.com/${pageId}`,
+            description,
+            item: items.filter((item) => !!item),
+        };
+    } else {
+        // fetch with mobile site fail (facebook return webpage without any story), try fetch with desktop site
+        const page = await ctx.cache.tryGet(
+            `fb_desktop_${pageId}`,
+            async () => {
+                const browser = await require('@/utils/puppeteer')();
+                const page = await browser.newPage();
+                await page.setViewport({ width: 390, height: 844 }); // iphone resolution
+                await page.goto(`https://www.facebook.com/${pageId}`, {
+                    waitUntil: 'networkidle0',
```

Comment:
Tests are failed in local environment as well probably because they get stuck here. 
Puppeteer didn't scroll down for me.

Maybe it's caused by background beacon/ping to `https://www.facebook.com/ajax/bz` or `https://www.facebook.com/nw/`.